Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,422 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Globalization;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        private const string currencySymbol = "¤";
+        private const string invalidMessage = "N/A.";
+        private const string messageEnd = ".";
+        private static readonly NumberFormatInfo numberFormatInfo = new NumberFormatInfo() { NumberDecimalSeparator = messageEnd };
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (!ArrayIsValid(peopleAndBalances)) return invalidMessage;
+
+            const string peopleMessage = " had the most money ever. ";
+            const string singlePersonMessage = " had the most money ever. ";
+
+            var resultMessage = new StringBuilder();
+            var peopleList = new string[peopleAndBalances.Length];
+            var highestBalanceList = new decimal[peopleAndBalances.Length];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                highestBalanceList[i] = HighestBalanceForSinglePerson(peopleAndBalances[i]);
+                peopleList[i] = ReturnNameForSinglePerson(peopleAndBalances[i]);
+            }
+
+            var highestBalanceIndex = FindDecimalArrayMax(highestBalanceList);
+            var highestBalance = highestBalanceList[highestBalanceIndex];
+            var personWithHighestBalance = peopleList[highestBalanceIndex];
+
+            if (ArrayElementsAreEqual(highestBalanceList))
+            {
+                resultMessage
+                    .Append(FormatStringAndCommas(peopleList))
+                    .Append(peopleMessage)
+                    .Append(FormatCurrency(highestBalance, currencySymbol))
+                    .Append(messageEnd);
+                return resultMessage.ToString();
+            }
+            else
+            {
+                resultMessage.Clear();
+
+                resultMessage
+                    .Append(personWithHighestBalance)
+                    .Append(singlePersonMessage)
+                    .Append(FormatCurrency(highestBalance, currencySymbol))
+                    .Append(messageEnd);
+
+                return resultMessage.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public static string FindMostPoorPerson(string[] peopleAndBalances)
+        {
+            if (!ArrayIsValid(peopleAndBalances)) return invalidMessage;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4970157)

Reply:
Fixed in [2db6c28](https://github.com/csinn/CSharp-From-Zero-To-Hero/pull/161/commits/2db6c282219933cc661146c542a210b0f73f44cf)